packages/@aws-cdk/aws-codedeploy/lib/ecs/deployment-group.ts

```diff
@@ -32,14 +46,144 @@ export interface IEcsDeploymentGroup extends cdk.IResource {
 }
 
 /**
- * Note: This class currently stands as a namespaced container for importing an ECS
- * Deployment Group defined outside the CDK app until CloudFormation supports provisioning
- * ECS Deployment Groups. Until then it is closed (private constructor) and does not
- * extend {@link Construct}.
+ * Details about an ECS service and its corresponding ECS cluster
+ */
+export type EcsService = ecs.Ec2Service | ecs.FargateService | ecs.ExternalService;
+
+/**
+ * A route with an ELB listener and target group.
+ */
+export interface EcsTrafficRoute {
+  /**
+   * The load balancer for the traffic route.
+   *
+   * @default - blank if empty on test traffic route. Required for prod traffic route.
+   */
+  readonly listener: elbv2.IApplicationListener | elbv2.INetworkListener | undefined;
+  /**
+   * The target group for the traffic route.
+   */
+  readonly targetGroup: elbv2.ITargetGroup;
+}
+
+/**
+ * Configuration for the blue/green CodeDeploy deployment.
+ */
+export interface EcsBlueGreenDeploymentConfig {
+  /**
+   * The time to wait for a ContinueDeployment after the deployment is ready.
+   * Leave blank to automatically continue once the deployment is ready.
+   *
+   * @default - automatically continue once deployment is ready
+   */
+  readonly waitTimeForContinueDeployment?: cdk.Duration;
+  /**
+   * The time to wait before terminating old tasks.
+   * Leave blank to keep the old tasks alive and deregistered from the load balancer.
+   *
+   * @default - keep the old tasks alive and deregistered from the load balancer.
```

Comment:
Ooh this seems like an unexpected and pricey default to let those old task sets hang around.  I seem to remember that the next deployment would fail if the task set is not terminated, because CodeDeploy requires there to be exactly one task set in the service when the deployment starts.  I would expect that the default would be 0 wait time for termination

Reply:
Turns out this interface was cruft - Removed.